app/controllers/friendships_controller.rb

```diff
@@ -0,0 +1,99 @@
+class FriendshipsController < ApplicationController
+  include LolConnections
+  before_action :search_summoner_for_request_action, only: [:edit]
+  before_action :require_user
+  before_action :require_same_user, only: [:edit, :decide]
+  
+ 
+	def new
+    @user = current_user
```

Comment:
You really need to look at how you are using instance variables in this controller.  Instance variables start with an @ sign.  You should keep the scope of a variable as tight as possible so if you don't need to use a variable outside of a particular function, then it should be local.  A lot of the variables here are only used in the function they are declared in so there is no reason to make them belong to the entire instance.  I'm not even sure how some of this is working correctly.  You redefine @user 4 times and you are even using the same definition for some of them.  
For example, if you wanted to get the current user you could do `@user ||= current_user`.  This means, it will only execute the current_user function if @user is not already defined.  You don't want to redo any calls that you don't have to.  Many of these (including `current_user`) require db queries and are not cheap to execute.  Reevaluate how you are using ALL your variables in this controller and make changes as you deem necessary.